lib/rex/ui/text/shell/history_manager.rb

```diff
@@ -56,8 +56,19 @@ def clear_readline
 
     def load_history_file(history_file)
       clear_readline
-
       if File.exist?(history_file)
+        if File.readlines(history_file).length > 2000
```

Comment:
Instead of checking the size of the history file and then truncating it by writing to it before reading it again in the `#load_history_file` method, what we should do is simply load in the last 2000 lines.

Then in the `#store_history_file` we should write the most recent 2000 lines of the history to the file. This would mean we'd be completely writing the history file instead of just appending to it.

This pattern would result in more data being written but would reduce the amount of logic necessary to ensure the size limit is consistently enforced which should really be handled at *both* boundaries (loading and storing).

Reply:
sounds better, will do that, thanks!